src/main/java/fi/helsinki/cs/tmc/cli/backend/Settings.java

```diff
@@ -116,8 +126,58 @@ public Path getConfigRoot() {
     }
 
     @Override
-    public void setCourse(Course course) {}
+    public String hostProgramName() {
+        // which command line is used
+        return null;
+    }
+
+    @Override
+    public String hostProgramVersion() {
+        // which command line is used
+        return null;
+    }
 
     @Override
-    public void setConfigRoot(Path path) {}
+    public boolean getSendDiagnostics() {
+        return false;
+    }
+
+    @Override
+    public Optional<OauthCredentials> getOauthCredentials() {
+        return account.getOauthCredentials();
+    }
+
+    @Override
+    public void setOauthCredentials(Optional<OauthCredentials> credentials) {
+        account.setOauthCredentials(credentials);
+    }
+
+    @Override
+    public void setToken(Optional<String> token) {
+        account.setoAuthToken(token);
+    }
+
+    @Override
+    public Optional<String> getToken() {
+        return account.getoAuthToken();
+    }
+
+    @Override
+    public Optional<Organization> getOrganization() {
+        return account.getOrganization();
+    }
+
+    @Override
+    public void setOrganization(Optional<Organization> organization) {
+        account.setOrganization(organization);
+    }
+
+    @Override
+    public void setCourse(Course course) {
+        account.setCurrentCourse(Optional.of(course));
+    }
+
+    @Override
+    public void setConfigRoot(Path path) {
```

Comment:
What is `setConfigRoot`? :confused:

Reply:
This has been discussed earlier and apparently one shouldn't be able to change the config root, so the method is pointless (NOP in tmc netbeans as well). It should be removed from the API.